### PR TITLE
ELEMENTS-860: change id function for handling doc refs and selection formatter

### PR DESCRIPTION
**Pull request by nmpcunha**
https://qa.nuxeo.org/jenkins/view/TestAndPush/job/web-ui-pipeline-ncunha/113/

### CODE REVIEW

**Comment by guirenard** [comment]
:+1:

File: widgets/nuxeo-document-suggestion.html [hunk 1]
```diff
@@ -272,6 +272,9 @@
         }
 
         _selectionFormatter(doc) {
+          if (!doc['entity-type'] && typeof doc === 'string') {
```
Comment (Gabez0r) [inline widgets/nuxeo-document-suggestion.html]: if doc is a string, then `!doc['entity-type']` will always be `true`:
```suggestion
          if (typeof doc === 'string') {
```

File: widgets/nuxeo-document-suggestion.html [hunk 2]
```diff
@@ -320,11 +323,10 @@
         }
 
         _docIdFunction(doc) {
-          if (this.idProperty === 'ecm:uuid') {
-            return doc.uid;
-          } else {
-            return doc.properties[this.idProperty];
+          if (!doc['entity-type'] && typeof doc === 'string') {
```
Comment (Gabez0r) [inline widgets/nuxeo-document-suggestion.html]: Same here:
```suggestion
          if (typeof doc === 'string') {
```

**Comment by Gabez0r** [comment]
👍

**Pull request merged**